### PR TITLE
GEODE-8149: New parameter and property introduced

### PR DESCRIPTION
**Pull request by jvarenina**
* new parameter security-cn-auth-enabled introduced
* new property in security-manager interface introduced

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/distributed/ConfigurationProperties.java [hunk 1]
```diff
@@ -2032,6 +2032,23 @@
...
    */
   String SECURITY_AUTH_TOKEN_ENABLED_COMPONENTS = SECURITY_PREFIX + "auth-token-enabled-components";
+
+  /**
+   * The static String definition of the <i>"security-cn-auth-enabled"</i> property
+   * </p>
+   *
+   * <U>Description</U> This parameter only works if SSL with two-way handshake is enabled on
+   * all components and security manager is enabled. This property will determine if common name
+   * from client certificate will be used for authentication and authorization.
```
Comment (davebarnes97) [inline geode-core/src/main/java/org/apache/geode/distributed/ConfigurationProperties.java:2042]: Present tense would be clearer: 
This property allows the common name from the client certificate to be used for authentication and authorization.

Comment (jvarenina) [reply geode-core/src/main/java/org/apache/geode/distributed/ConfigurationProperties.java:2042]: Thanks for the comments. I will rephrase it as you suggested.

**Comment by jvarenina** [comment]
> This PR provides documentation at the Javadoc level.
> Consider where in the User Guide this should be mentioned, then
> 
>     * add a User Guide explanation as part of this PR/JIRA combination, or
> 
>     * add a subticket to GEODE-8118, the parent ticket of this one (GEODE-8149) to handle the User Guide docs.
>       Either of the above will satisfy my concern. Thanks.

OK, I will create sub-ticket to handle User Guide docs.

**Comment by davebarnes97** [comment]
> OK, I will create sub-ticket to handle User Guide docs.
I'm not finding the sub-ticket for docs. Please provide the ticket number - thanks.

**Comment by jvarenina** [comment]
Hi,
Yes, property is introduced for future works.
The reason I didn't update this PR is because we faced some issues when developing this feature which could affect this PR greatly when resolved. We have problem with retrieving certificates from RMI connections(e.g. jmx connections) and I will send more info related to this issue in dev list. We have also put on hold all other task related to this feature until we find the solution for RMI Connections. 
@davebarnes97 thanks again for comments I will try to implement them as soon as possible.

**Comment by pivotal-jbarrett** [comment]
@jvarenina How about you close this PR until that is resolved then.

**Pull request closed without merging**